### PR TITLE
Nuke tgui port

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Ports https://github.com/Cosmic-Overlord/TerraGov-Marine-Corps/pull/41

image

Also made removing nuke disks insert into your hand instead of just flop onto the floor under the nuke.
## Why It's Good For The Game
TGUI is nice
## Changelog
:cl:
add: Added TGUI to the nuke
qol: Nuke disks are inserted into your hand, when removed from the nuke

/:cl:

### CODE REVIEW

File: code/game/objects/machinery/nuclearbomb.dm [hunk 1]
```diff
@@ -205,125 +213,134 @@
...
+/obj/machinery/nuclearbomb/proc/toggle_timer()
+	if(has_auth)
+		if(exploded)
+			return
+		if(safety)
+			to_chat(usr, span_warning("The safety is still on."))
+			return
+		if(!anchored)
+			to_chat(usr, span_warning("The anchors are not set."))
+			return
```
Comment (TiviPlus) [inline code/game/objects/machinery/nuclearbomb.dm:249]: usr + balloon alerts

File: code/game/objects/machinery/nuclearbomb.dm [hunk 2]
```diff
@@ -205,125 +213,134 @@
...
+		if("toggle_timer")
+			toggle_timer()
+		if("change_time")
+			change_time(params["seconds"])
+		if("toggle_safety")
+			toggle_safety()
+		if("toggle_anchor")
+			toggle_anchor()
+		if("toggle_disk")
+			toggle_disk(params["disktype"])
```
Comment (TiviPlus) [inline code/game/objects/machinery/nuclearbomb.dm:237]: hasauth should be checked here and not in the individual procs for toggling

Comment (TiviPlus) [inline code/game/objects/machinery/nuclearbomb.dm:237]: I dont see sanitizing for the params here

File: code/game/objects/machinery/nuclearbomb.dm [hunk 3]
```diff
@@ -205,125 +213,134 @@
...
+///Toggles the anchor bolts on or off
+/obj/machinery/nuclearbomb/proc/toggle_anchor()
+	if(has_auth)
+		if(removal_stage == NUKE_STAGE_BOLTS_REMOVED)
+			anchored = FALSE
+			visible_message(span_warning("\The [src] makes a highly unpleasant crunching noise. It looks like the anchoring bolts have been cut."))
+			return
+		if(istype(get_area(loc), /area/shuttle))
+			to_chat(usr, span_warning("This doesn't look like a good spot to anchor the nuke."))
+			return
```
Comment (TiviPlus) [inline code/game/objects/machinery/nuclearbomb.dm:283]: more usr and tochat not going to repeat myself for the rest

File: code/game/objects/machinery/nuclearbomb.dm [hunk 4]
```diff
@@ -205,125 +213,134 @@
...
+	if(has_auth)
+		safety = !safety
+		if(safety)
+			timer_enabled = FALSE
+			stop_processing()
+
+///Toggles the anchor bolts on or off
+/obj/machinery/nuclearbomb/proc/toggle_anchor()
+	if(has_auth)
+		if(removal_stage == NUKE_STAGE_BOLTS_REMOVED)
```
Comment (TiviPlus) [inline code/game/objects/machinery/nuclearbomb.dm:277]: lot of missing early returns

File: code/game/objects/machinery/nuclearbomb.dm [hunk 5]
```diff
@@ -205,125 +213,134 @@
...
+			if("blue")
+				b_auth = I
+		if(r_auth && g_auth && b_auth)
+			has_auth = TRUE
 
+/obj/machinery/nuclearbomb/interact(mob/user)
+	. = ..()
+	if(.)
+		return
+	ui_interact(user)
```
Comment (TiviPlus) [inline code/game/objects/machinery/nuclearbomb.dm:341]: We have flags for this unnecessary code

**Review by TiviPlus — CHANGES_REQUESTED** [review]
pics?

File: code/game/objects/machinery/nuclearbomb.dm (hunk 2 above)
Comment (Lumipharon) [reply code/game/objects/machinery/nuclearbomb.dm:237]: done

File: code/game/objects/machinery/nuclearbomb.dm (hunk 1 above)
Comment (Lumipharon) [reply code/game/objects/machinery/nuclearbomb.dm:249]: done

File: code/game/objects/machinery/nuclearbomb.dm (hunk 2 above)
Comment (Lumipharon) [reply code/game/objects/machinery/nuclearbomb.dm]: isnum works for "seconds", but what sanitisation should ther be for "disktype"?

File: code/game/objects/machinery/nuclearbomb.dm (hunk 5 above)
Comment (Lumipharon) [reply code/game/objects/machinery/nuclearbomb.dm:341]: done

File: code/game/objects/machinery/nuclearbomb.dm (hunk 4 above)
Comment (Lumipharon) [reply code/game/objects/machinery/nuclearbomb.dm]: done as far as I can see

File: code/game/objects/machinery/nuclearbomb.dm (hunk 3 above)
Comment (Lumipharon) [reply code/game/objects/machinery/nuclearbomb.dm]: done

**Review by TiviPlus — APPROVED** [review]
Someone should clean up the old code at some point

**Pull request merged**